crates/graph/src/main.rs

```diff
@@ -0,0 +1,61 @@
+//! Actix web juniper example
+//!
+//! A simple example integrating juniper in actix-web
+use std::io;
+use std::sync::Arc;
+
+use actix_cors::Cors;
+use actix_web::{middleware, web, App, Error, HttpResponse, HttpServer};
+use juniper::http::graphiql::graphiql_source;
+use juniper::http::GraphQLRequest;
+
+mod schema;
+
+use crate::schema::{create_schema, Schema};
+
+async fn graphiql() -> HttpResponse {
+    let html = graphiql_source("http://127.0.0.1:3000", None);
+    HttpResponse::Ok()
+        .content_type("text/html; charset=utf-8")
+        .body(html)
+}
+
+async fn graphql(
+    st: web::Data<Arc<Schema>>,
+    data: web::Json<GraphQLRequest>,
+) -> Result<HttpResponse, Error> {
+    let json = web::block(move || {
+        let res = data.execute_sync(&st, &());
+        serde_json::to_string(&res)
+    })
+    .await?;
+    Ok(HttpResponse::Ok()
+        .content_type("application/json")
+        .body(json))
+}
+
+#[actix_web::main]
+async fn main() -> io::Result<()> {
+    std::env::set_var("RUST_LOG", "actix_web=info");
+    env_logger::init();
+
+    let schema = std::sync::Arc::new(create_schema());
+
+    HttpServer::new(move || {
+        App::new()
+            .data(schema.clone())
+            .wrap(middleware::Logger::default())
+            .wrap(
+                Cors::new()
+                    .allowed_methods(vec!["POST", "GET"])
+                    .supports_credentials()
+                    .max_age(3600)
+                    .finish(),
+            )
+            .service(web::resource("/").route(web::post().to(graphql)))
+            .service(web::resource("/graphiql").route(web::get().to(graphiql)))
+    })
+    .bind("127.0.0.1:3000")?
+    .run()
+    .await
+}
```

Comment:
This should pull in `clap` and have a similar argument parser to the `rpc` crate for allowing the port to be changed.  Additionally I have already set up of some of the boilerplate written here in the `indexer_core` crate.  I can look into making sure it works with async, but I would much prefer to reuse that than pulling in the same dependencies in a slightly different manner.